executor/executor_test.go

```diff
@@ -1102,6 +1102,18 @@ func (s *testSuite) TestUnion(c *C) {
 	for i := 0; i < 4; i++ {
 		tk.MustQuery("SELECT(SELECT 0 AS a FROM dual UNION SELECT 1 AS a FROM dual ORDER BY a ASC  LIMIT 1) AS dev").Check(testkit.Rows("0"))
 	}
+
+	// #issue 8196
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("drop table if exists t2")
+	tk.MustExec("CREATE TABLE t1 (a int not null, b char (10) not null)")
+	tk.MustExec("insert into t1 values(1,'a'),(2,'b'),(3,'c'),(3,'c')")
+	tk.MustExec("CREATE TABLE t2 (a int not null, b char (10) not null)")
+	tk.MustExec("insert into t2 values(3,'c'),(4,'d'),(5,'f'),(6,'e')")
+	tk.MustExec("analyze table t1")
+	tk.MustExec("analyze table t2")
+	_, err = tk.Exec("(select a,b from t1 limit 2) union all (select a,b from t2 order by a limit 1) order by t1.b")
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 't1.b' in 'order clause'")
```

Comment:
In order to do that, we should know the statement type (e.g., ``SELECT``). If we can confirm that ``UNION`` is ``SELECT`` statement, we can generate the same error like MySQL for this case only, which might be ad hoc. The generalized error message handling might beyond the PR.

Reply:
Why do we need to know whether the statement type?
Can `OrderByClause.ForUnion` help?

Reply:
Seems `ForUnion` is never set now, we may need parser change.

Reply:
The updated PR should produce the same error message like MySQL.

Reply:
Hi, I do not really understand why don't we use ForUnion? @dbjoa

Reply:
@XuHuaiyu We can use ``ForUnion`` to indicate ``OrderByClause`` used in ``UNION`` statement. However, I have concern that to use the indicator the existing methods should change the files in the call stack significantly comparing to the PR.  For example. ``PlanBuilder.buildSort()`` needs to use ``ast.OrderByClause`` as its argument instead of ``ast.ByItem``, and accordingly, ``expression.expressionRewriter()`` needs to extend for rewriting ``ast.OrderByClause".

Reply:
Can we use `ForUnion` and check if table name is specified in `Preprocess`? We can report error there, so we don't need to modify `buildSort` and `expressionRewriter`?

Reply:
@eurekaka Yes, we can. However, IMHO, it might not be a simple work as done in the PR because we need to write codes to check all the constraints as done in the existing implementation to produce the same errors.